File: ipam/backend/app/schemas/site.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class SiteBase(BaseModel):
    name: str
    slug: str
    status: str
    region_id: Optional[int] = None
    site_group_id: Optional[int] = None

class SiteCreate(SiteBase):
    pass

class SiteUpdate(SiteBase):
    name: Optional[str] = None
    slug: Optional[str] = None
    status: Optional[str] = None

class SiteResponse(SiteBase):
    id: int
    created_at: datetime

    class Config:
        orm_mode = True

File: ipam/frontend/src/types/site.ts
export interface Site {
  id: number;
  name: string;
  slug: string;
  status: string;
  region_id?: number;
  site_group_id?: number;
  created_at: string;
}

export interface CreateSiteData {
  name: string;
  slug: string;
  status: string;
  region_id?: number;
  site_group_id?: number;
}

export interface UpdateSiteData extends Partial<CreateSiteData> {}
